sequencer/sequencer.go

```diff
@@ -280,3 +347,31 @@ func (s sequencer) sendControls() {
 		track.sendControls()
 	}
 }
+
+// Helper functions for deep copying pointers
+func copyIntPtr(ptr *int) *int {
+	if ptr == nil {
+		return nil
+	}
+	copy := *ptr
+	return &copy
+}
+
+func copyUint8Ptr(ptr *uint8) *uint8 {
+	if ptr == nil {
+		return nil
+	}
+	copy := *ptr
+	return &copy
+}
+
+func copyUint8SlicePtr(ptr *[]uint8) *[]uint8 {
+	if ptr == nil {
+		return nil
+	}
+	copy := make([]uint8, len(*ptr))
+	for i, v := range *ptr {
+		copy[i] = v
+	}
+	return &copy
+}
```

Comment:
Could you move these in a separate `helper.go` file? Let's keep it in the sequencer package for now.